### PR TITLE
Improvements to app documentation

### PR DESCRIPTION
**Pull request by mikewilburn**
This PR:
- renames `docs/index.md` to `docs/README.md`
- adds table of contents to (2) README files. The TOC was generated using MDTOC.
- links to `docs/README.md` from root README.md
- makes small formatting changes to root README.md (d53e001, 7c4bf52, & 607c171)
- increments app version patch number
- describes patch changes in RELEASE.md

@puneet-pdx can you tell me if I caught all the locations to increment the app version number?

### CODE REVIEW

File: README.md
Comment (esreli) [inline README.md:138]: Good catch

File: app/src/main/res/values/strings.xml
Comment (esreli) [inline app/src/main/res/values/strings.xml:4]: Interesting, we should resolve our differences here. I thought changing the version number was done within the app's [gradle](https://github.com/Esri/mapbook-android/blob/fb9319742dbc205873a3f22f6e47eccecbf1b5fc/mapbook-app/build.gradle#L10) file. @puneet-pdx can you weigh in here?

**Review by esreli — APPROVED** [review]
Other than one note about the app's version number this PR looks good to me! @puneet-pdx will have to make final judgement on the app version number question.

File: app/src/main/res/values/strings.xml
Comment (mikewilburn) [reply app/src/main/res/values/strings.xml:4]: @esreli I thought so too based on your other PR, but in the [`build.gradle`](https://github.com/Esri/data-collection-android/blob/master/app/build.gradle#L17) file, it's simply noted as `"1.0"` and doesn't look like it affects what's written out to the UI.

@puneet-pdx your insight here would be helpful.

**Comment by puneet-pdx** [comment]
@mikewilburn should't we be merging this PR in `v.next` instead of `master` ?

File: app/src/main/res/values/strings.xml
Comment (puneet-pdx) [reply app/src/main/res/values/strings.xml:4]: The app currently is not reading the VersionName from `build.gradle` file when we are displaying it in the UI. So Unfortunately we will have to do the duplication here. I can fix this in a separate PR.

Comment (puneet-pdx) [reply app/src/main/res/values/strings.xml:4]: The versionName needs to be updated here as well https://github.com/Esri/data-collection-android/blob/master/app/build.gradle#L17

**Review by puneet-pdx — CHANGES_REQUESTED** [review]
@mikewilburn should't we be merging this PR in v.next instead of master ?

**Comment by esreli** [comment]
https://github.com/Esri/data-collection-android/pull/5#pullrequestreview-458706856

@puneet-pdx Because this is a patch release, it should soon make it's way into `master` and be released as a patch. Do you have other work you'd like to slate into a near-term patch release? 

What git flow do you practice here? It was my impression that `v.next` was for `0.2`, is this incorrect?

**Comment by mikewilburn** [comment]
@puneet-pdx Eli is right - we should merge this patch fix directly into `master`.

After this PR is merged, you can either rebase `v.next` to include `master`'s latest changes, or simply resolve any conflicts when it comes time to merge `v.next`.

I've also just thrown in another commit. Would you have a look and approve this if all looks good to you?

**Review by puneet-pdx — APPROVED** [review]
LGTM

**Pull request merged**
